dimod/generators/satisfiability.py

```diff
@@ -0,0 +1,137 @@
+        yield (y, z, ys*zs)
+
+
+def _iter_interactions_without_replacement(
+        num_variables: int, num_clauses: int,
+        seed: typing.Union[None, int, np.random.Generator] = None,
+        ) -> typing.Iterator[typing.Tuple[int, int, int]]:
+
+    try:
+        max_num_clauses = math.comb(num_variables, 3)
+    except AttributeError:
+        # Python < 3.8
+        # This is a bit less picky and has overflow issues but
+        # probably fine for any cases users are likely to actually
+        # submit
+        max_num_clauses = math.factorial(num_variables)
+        max_num_clauses //= (6 * math.factorial(num_variables - 3))
+
+    if num_clauses > max_num_clauses:
+        raise ValueError("rho results in too many clauses")
+
+    rng = np.random.default_rng(seed)
+
+    seen: typing.Set[typing.FrozenSet[int]] = set()
+    while len(seen) < num_clauses:
+        x, y, z = rng.choice(num_variables, 3, replace=False)
+
+        fz = frozenset((x, y, z))  # so we're order-invarient
+        if fz in seen:
+            continue
+        seen.add(fz)
+
+        xs, ys, zs = 2 * rng.integers(0, 1, endpoint=True, size=3) - 1
+
+        yield (x, y, xs*ys)
+        yield (x, z, xs*zs)
+        yield (y, z, ys*zs)
+
+
+def nae3sat(variables: typing.Union[int, typing.Sequence[dimod.typing.Variable]],
+            rho: float = 2.1,
+            *,
+            seed: typing.Union[None, int, np.random.Generator] = None,
+            replace: bool = True,
+            ) -> BinaryQuadraticModel:
+    """Generator for Not-All-Equal 3-SAT (NAE3SAT) Binary Quadratic Models.
+
+    NAE3SAT_ is an NP-complete problem class that consists in satistying a number of conjunctive
+    clauses that involve three variables (or variable negations). The variables on each clause
+    should be not-all-equal. Ie. all solutions except ``(+1, +1, +1)`` or
+    ``(-1, -1, -1)`` are valid for each class.
+
+    .. _NAE3SAT: https://en.wikipedia.org/wiki/Not-all-equal_3-satisfiability
+
+    Args:
+        num_variables: The number of variables in the problem.
+        rho: The clause-to-variable ratio.
+        seed: Passed to :func:`numpy.random.default_rng()`, which is used
+            to generate the clauses and the negations.
+        replace: If true, then clauses are randomly sampled from the space
+            of all possible clauses. This can result in the same three variables
+            being present in multiple clauses.
+            As the number of variables grows the probability of this happening
+            shrinks rapidly and therefore it is often better to allow sampling
+            with replacement for performance.
+
+    Returns:
+        A binary quadratic models with spin variables.
+
+    """
+    if isinstance(variables, collections.abc.Sequence):
+        num_variables = len(variables)
+        labels = variables
+    elif variables < 0:
+        raise ValueError("variables must be a sequence or a positive integer")
```

Comment:
```suggestion
    elif variables < 3:
        raise ValueError("variables must be a sequence or a positive integer of at least size 3")
```

Currently you can set a size of 2 and you'll get some unhelpful error from ``rng.choice()``